Content.Shared/Interaction/SharedInteractionSystem.cs

```diff
@@ -240,7 +240,12 @@ private bool ShouldCheckAccess(EntityUid user)
         {
             if (TryComp<InteractionRelayComponent>(user, out var relay) && relay.RelayEntity is not null)
             {
-                UserInteraction(relay.RelayEntity.Value, coordinates, target, altInteract, checkCanInteract, checkAccess, checkCanUse);
+                // TODO this needs to be handled better. This probably bypasses many complex can-interact checks in weird roundabout ways.
+                if (_actionBlockerSystem.CanInteract(user, target))
+                {
+                    UserInteraction(relay.RelayEntity.Value, coordinates, target, altInteract, checkCanInteract, checkAccess, checkCanUse);
+                    return;
+                }
```

Comment:
specifically: this return here might prevent the mech pilot from performing some interactions while inside a mech, but I'm not sure what those might be.